source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,292 @@
+{
+    mCullCreaturesFlag = true;
+}
+
+void CullingManager::startTileCulling()
+{
+    getIntersectionPoints();
+
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+    mOldWalk = mWalk;
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+void CullingManager::startDebugging()
+{
+    mDebug = true;
+}
+
+void CullingManager::stopCreatureCulling()
+{
+    mCullCreaturesFlag = false;
+}
+
+void CullingManager::stopTileCulling()
+{
+    mCullTilesFlag = false;
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mWalk.debug());
+        
+    }
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->hide();
+        }
+    }
+}
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->show();
+        }
+    }
+}
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = mOldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= mOldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , mOldWalk.getBottomRightVertex().y) >> mPrecisionDigits) - 2) << mPrecisionDigits;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , mOldWalk.getTopRightVertex().y  ) >> mPrecisionDigits) + 2) << mPrecisionDigits;  yy >= bb; yy -= Unit)
+    {
+        mOldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        xxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeftOld = mOldWalk.getCurrentXLeft(yy);
+        xxRight = mWalk.getCurrentXRight(yy);
+        xxRightOld = mOldWalk.getCurrentXRight(yy);
+   
+        // if(mDebug)
```

Comment:
Please do not let dead code. If there is a need, add a debug log. If not, just remove it.

BTW, you should not use streams for logs. It is much better to use OD_LOG_DBG macro because if the log level is not high enough, the message will not be computed.